src/evm/protocol/vm/state.rs

```diff
@@ -599,11 +604,107 @@ mod tests {
                 "src/evm/protocol/vm/assets/BalancerSwapAdapter.evm.runtime".to_string(),
             ))
             .stateless_contracts(stateless_contracts)
-            .build()
+            .build(SHARED_TYCHO_DB.clone())
             .await
             .expect("Failed to build pool state")
     }
 
+    #[ignore]
+    #[rstest]
+    fn test_get_amount_out_sim_db() {
+        // TODO remove this test after review - this is just to show what I've tried.
+        // TEST SETUP - fails due to
+        // "there is no reactor running, must be called from the context of a Tokio 1.x runtime"
+
+        let data_str = include_str!("assets/balancer_contract_storage_block_20463609.json");
+        let data: Value = serde_json::from_str(data_str).expect("Failed to parse JSON");
+
+        let accounts: Vec<AccountUpdate> = serde_json::from_value(data["accounts"].clone())
+            .expect("Expected accounts to match AccountUpdate structure");
+
+        let rpc_url = std::env::var("ETH_RPC_URL").unwrap();
+        let client = Provider::<Http>::try_from(rpc_url).unwrap();
+        let runtime = Arc::new(tokio::runtime::Runtime::new().unwrap());
+        let client = Arc::new(client);
+        let db = SimulationDB::new(client, None, None);
```

Comment:
We should be passing the runtime here. Unfortunately we can't do that without moving it, and we need this very runtime below in order to build the pool state.

When we don't pass the runtime here, we the problems:
```
there is no reactor running, must be called from the context of a Tokio 1.x runtime
```